core/object/message_queue.h

```diff
@@ -89,13 +90,7 @@ class CallQueue {
 		};
 	};
 
-	_FORCE_INLINE_ void _ensure_first_page() {
-		if (unlikely(pages.is_empty())) {
-			pages.push_back(allocator->alloc());
-			page_bytes.push_back(0);
-			pages_used = 1;
-		}
-	}
+	_FORCE_INLINE_ void _ensure_first_page();
```

Comment:
I think `_FORCE_INLINE_` won't work on forward declarations (though I might be wrong).

```suggestion
	void _ensure_first_page();
```

Reply:
It seems to work. Try this for example: https://godbolt.org/z/6qMnsoT7E

Reply:
I don't see how this can work. To inline something, the compiler needs to know the definition. It works in your Compiler Explorer example because the declaration and definition are in the same file, which is not the case in this PR.

Reply:
Ah, I was thinking about the case where it would be called in a compilation unit different from the one where it is defined, but it seems that isn't the case here. Interesting!

Reply:
In that case the code would actually not compile, in my experience. The compiler would generate duplicate definitions for the function, causing a compile error.

Reply:
I'm not surprised `_FORCE_INLINE_` works on split declaration / definition sites. What I was personally wondering about was whether it was needed on the declaration site _in addition to_ the definition site.
Since the function isn't called outside the compilation unit, it doesn't really matter practically. For me it was mostly about not needing to expose whether it was `_FORCE_INLINE_` or not to other users of the header.